bridges/snowbridge/runtime/test-common/src/lib.rs

```diff
@@ -193,12 +195,96 @@ pub fn send_transfer_token_message_success<Runtime, XcmConfig>(
 
 			let digest = included_head.digest();
 
-			//let digest = frame_system::Pallet::<Runtime>::digest();
 			let digest_items = digest.logs();
 			assert!(digest_items.len() == 1 && digest_items[0].as_other().is_some());
 		});
 }
 
+pub fn test_pallet_order_works<Runtime, XcmConfig, AllPalletsWithoutSystem>(
+	collator_session_key: CollatorSessionKeys<Runtime>,
+	runtime_para_id: u32,
+	assethub_parachain_id: u32,
+	weth_contract_address: H160,
+	destination_address: H160,
+	fee_amount: u128,
+	snowbridge_pallet_outbound_queue: Box<
+		dyn Fn(Vec<u8>) -> Option<snowbridge_pallet_outbound_queue::Event<Runtime>>,
+	>,
+) where
+	Runtime: frame_system::Config
+		+ pallet_balances::Config
+		+ pallet_session::Config
+		+ pallet_xcm::Config
+		+ parachain_info::Config
+		+ pallet_collator_selection::Config
+		+ pallet_message_queue::Config
+		+ cumulus_pallet_parachain_system::Config
+		+ snowbridge_pallet_outbound_queue::Config
+		+ snowbridge_pallet_system::Config
+		+ pallet_timestamp::Config,
+	XcmConfig: xcm_executor::Config,
+	AllPalletsWithoutSystem:
+		OnInitialize<BlockNumberFor<Runtime>> + OnFinalize<BlockNumberFor<Runtime>>,
+	ValidatorIdOf<Runtime>: From<AccountIdOf<Runtime>>,
+	<Runtime as frame_system::Config>::AccountId: From<sp_runtime::AccountId32> + AsRef<[u8]>,
+{
+	ExtBuilder::<Runtime>::default()
+		.with_collators(collator_session_key.collators())
+		.with_session_keys(collator_session_key.session_keys())
+		.with_para_id(runtime_para_id.into())
+		.with_tracing()
+		.build()
+		.execute_with(|| {
+			<snowbridge_pallet_system::Pallet<Runtime>>::initialize(
+				runtime_para_id.into(),
+				assethub_parachain_id.into(),
+			)
+			.unwrap();
+
+			// fund asset hub sovereign account enough so it can pay fees
+			initial_fund::<Runtime>(assethub_parachain_id, 5_000_000_000_000);
+
+			let outcome = send_transfer_token_message::<Runtime, XcmConfig>(
+				assethub_parachain_id,
+				weth_contract_address,
+				destination_address,
+				fee_amount,
+			);
+
+			assert_ok!(outcome.ensure_complete());
+
+			// check events
+			let mut events = <frame_system::Pallet<Runtime>>::events()
```

Comment:
@claravanstaden if you add MessageQueue before EthereumOutboundQueue, what is the exact issue/problem/error? Does MessageQueue or anything else trigger any other Event or something? I mean, is it possible to add a check for that negative scenario? E.g. `System::events()` does not contain `MessageQueue::Processed` or whatever?

Reply:
The MessageQueue::Processed event will still be there, but `Other` digest item will not be found in the DigestLogs. https://github.com/paritytech/polkadot-sdk/pull/3381/files/58729e09808a758c079a0dd5334718f1db45adbd#diff-6abc523c7c31eb7defa80d6c139fd2b2a7de5b299da9f9bfeebdb2748881886cR279

I thought about adding a negative test, but I thought it didn't really make sense because that wouldn't use the bridge hub runtime and wouldn't catch the actual runtime misconfiguration. I did however test that this test fails if the order is incorrect.

Reply:
ok, got it, so there is nothing (Event, DigestItem,..) to check for case when MessageQueue processes first?

Reply:
Exactly.